Fix block validation error message

atimmer opened the pull request:
## Description

The `Actual` and `Expected` values were the wrong way around. This made the warning very confusing. The first commit renames the variables so the bug is made more clear. The second commit then fixes the bug by rearranging the variables in the `log.error` call.

## How has this been tested?

By trying to create a migration where the result of the `save` function was the empty string. Then it was very obvious because the value after `Expected` was empty.

## Types of changes

Bug fix.

atimmer committed:
Rename variable to make code more clear

atimmer committed:
Fix display of expected and actual block HTML

The variable were in the wrong order, making the message confusing. It
would show the actual HTML as the expected HTML and visa versa.

jorgefilipecosta reviewed (commented):
Hi @atimmer thank you for submitting this PR.

I did the following test:
I pasted this invalid gallery in the code editor:
```
<!-- wp:gallery {"ids":[22424],"className":"columns-1"} -->
<jhmhjk class="wp-block-gallery columns-1 is-cropped"><li class="blocks-gallery-item"><figure><img src="http://yah.local/wp-content/uploads/2019/02/300.jpg" alt="" data-id="22424" data-link="http://yah.local/?attachment_id=22424" class="wp-image-22424"/><figcaption>Wow caption parsed from metadata</figcaption></figure></li></ul>
<!-- /wp:gallery -->
```

I saved the post I reloaded and I got the following error:
![image](https://user-images.githubusercontent.com/11271197/54047562-4600bd00-41cf-11e9-9ed3-74ddb80585a2.png)


Where the expected and actual messages are swapped.

mmtr commented:
Hi @atimmer! I gave this a spin today and I'm not sure if I'm missing something. To me, the current implementation looks correct.

Currently, `isValidBlockContent` expects to receive the actual block content (the one that is saved on the post content) as third argument (`A`) and it compares it with the expected output that the current `save` method of the block produces (`B`).

That's why on the log we refer to `A` as `Actual` and `B` as `Expected`.

Maybe there is a specific scenario for migrating blocks that makes this confusing?

gziolo commented:
Either way it's confusing as is, we probably should update messages instead to make it crystal clear what was read from the post content and what was generated by the `save` implementation used for validation. It seems like what @atimmer tries to change makes sense but message is still confusing :)

atimmer commented:
After this brief exchange I understand what goes wrong here. `Expected` and `Actual` have a very specific meaning, which means when reading this a developer has a very specific expectation of these words.

Given my experience with unit tests and the fact that I was developing a `save` method this was really confusing. I expected the `save` function to be the `Actual` value, because that is what I was editing.

I agree with @gziolo that we should make this more clear by changing the language.

atimmer committed:
Change wording to be more clear

Expected & Actual were confusing words to use in this content. This
change makes the message actually reflect what the values are.

atimmer commented:
I've pushed a commit that should make this message clearer.

gziolo reviewed (approved):
Yes, this should make it easier to understand for everyone. Thanks a lot!

aduth commented:
You need to regenerate docs via `npm run docs:build`.

atimmer committed:
Regenerate docs

aduth commented on packages/blocks/src/api/validation.js:
```diff
@@ -620,30 +620,30 @@ export function isEquivalentHTML( actual, expected ) {
 	if ( ! isValid ) {
 		log.error(
-			'Block validation failed for `%s` (%o).\n\nExpected:\n\n%s\n\nActual:\n\n%s',
+			'Block validation failed for `%s` (%o).\n\nContent generated by `save` function:\n\n%s\n\nContent retrieved from post body:\n\n%s',
```
There shouldn't be a concept of a "post" in this context. This logic applies for the post editor, but also any blocks parsing (e.g. upcoming widgets screen blocks editor). I realize the desire to improve messaging to developers. There may be alternatives which serve as an adequate compromise:

```
Markup produced by `save` function:

...

Markup parsed from content:

...
```
